src/adagrams.js

```diff
@@ -1,15 +1,143 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  
+  
+
+  const letters = Object.keys(letterPool);
+  let hand = [];
+  while (hand.length < 10) {
```

Comment:
Nice job creating the hand! I'm not seeing where you account for the distribution of letters though. Seems like all letters have an equal chance of being selected.

Reply:
Sorry to seem like I'm being sensitive over this because I see I you still gave me green on this assignment and I appreciate it, but I really thought I got it right. I even ran to test 10 times in a row to make sure it worked. To my understanding, while const object keys are immutable their values aren't, so every time the loop ran can choose a letter from the list of keys it would go back and subtract from the value of that letter in the letter pool. The letter only got added to the hand if its matching value was above 0. Maybe I didn't do line 37 correctly. Please let me know.

Reply:
No problem! The function is getting letters from `letterPool` just fine, but it isn't accounting for the distribution of letters, so it should be the case that because the letter pool should contain 9 `I`s and 1 `J`, that it's 9 times more likely to draw an `I`. Think of it in an array form, simplified: 
```
['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'B', 'B',...]
```
If we generated a random index based on the length of this array, it would be much more likely to be an `A`. But yours works a bit differently. You generate a number between 1 and 26 and that is used as an index to key off of the letterPool object, so all letters have a 1 in 26 chance of being selected, whereas in the array above, if we created an array that had all the letters according to their distribution, which would produce an array of size 98, we'd end up with a 9/98 chance of drawing an A. As for the decrementing,  you're right that you need to decrement the value of each letter in the letter pool to make sure there are available tiles to draw and put into the hand, but there's a mismatch between `letters` and `letterPool` which is why the distribution of letters isn't accurate.